qpmodel/subquery.cs

```diff
@@ -627,13 +627,126 @@ public PhysicMarkJoin(LogicMarkJoin logic, PhysicNode l, PhysicNode r) : base(lo
 
         // find the #marker from children output and located the #marker by subquery_id_.
         void fixMarkerValue(Row r, Value value)
+        {
+            int ordinal = findMarkerOrdinal();
+            r[ordinal] = value;
+        }
+
+        int findMarkerOrdinal()
         {
             var output = this.logic_.output_;
             int ordinal = output.FindIndex(x => x is ExprRef xE && xE.child_() is MarkerExpr xEM && logic_ is LogicMarkJoin lm && xEM.subqueryid_ == lm.subquery_id_);
-            r[ordinal] = value;
+            return ordinal;
         }
 
+
+        private bool colsHasNull(Row r)
+        {
+            for (int i = 0; i < r.ColCount(); i++)
+            {
+                if (r[i] is null)
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private bool filterHasMarkerBinExpr(Expr filter) => filter.FilterToAndList().Exists(x => x is BinExpr xB && xB.IsMarkerBinExpr());
+
         public override void Exec(Action<Row> callback)
+        {
+            var isInMarkJoin = filterHasMarkerBinExpr(logic_.filter_);
+            if (isInMarkJoin)
+                ExecIn(callback);
+            else
+                ExecNotIn(callback);
+        }
+        public void ExecIn(Action<Row> callback)
```

Comment:
Are you implementing a semi-markjoin?

Reply:
> Can you specify what exactly not handled? Test cases (not handled before and now handled)?
scheme:
```

                // table a,b
                // 0 1 2 3
                // 1 2 3 4
                // 2 3 4 5
                // 3 N 4 5
```
The two queries.
```
// NULL not  in {} = true
TU.ExecuteSQL("SELECT a1 FROM a WHERE a1 = 3 and a2 NOT IN (SELECT b2 FROM b WHERE a1 < b1)", "3");
 // 3 NOT IN [1,2,NULL] = NULL
TU.ExecuteSQL("SELECT a1 FROM a WHERE a1 = 2 and a2 NOT IN (SELECT b2 FROM b WHERE b1 > a1 or b1 < a1)", "");
```
For the first query, the a1<b1 will fiter all b2, so it is a empyty set. say S = {}. When S = {}, it has result below. 
![image](https://user-images.githubusercontent.com/37564807/98490222-cfea0e80-226b-11eb-96bf-9a6e256bd1a2.png)
So, Null in {} = false, Null not in {} = true.
Fo1r the second query, the original PhysicMarkjoin will judge **3 in [1,2,NULL]** as false, so it will see **3 NOT IN [1,2,NULL]** as true. Infact, 3 in [1,2,NULL] is NULL and **3 NOT in [1,2,NULL]** is NULL too.

Reply:
try to merge left semi mark join and left mark join